File: uds-service/src/main/java/ru/uds/common/dto/ErrorResponse.java
package ru.uds.common.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import org.springframework.http.HttpStatus;

import java.util.Date;

@Data
public class ErrorResponse {
    private String message;
    private int code;
    private String status;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "MM-dd-yyyy hh:mm:ss")
    private Date timestamp = new Date();

    public ErrorResponse(HttpStatus httpStatus, String message) {
        this.code = httpStatus.value();
        this.status = httpStatus.name();
        this.message = message;
    }
}

File: tsodd-service/src/main/kotlin/ru/thetenzou/tsoddservice/common/dto/ErrorResponse.kt
package ru.thetenzou.tsoddservice.common.dto

import com.fasterxml.jackson.annotation.JsonFormat
import org.springframework.http.HttpStatus
import java.util.*

/**
 * Error response structure
 *
 * @param httpStatus response status
 * @param message text of the message
 */
class ErrorResponse(
    httpStatus: HttpStatus,
    val message: String,
) {

    val code: Int
    val status: String

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "MM-dd-yyyy hh:mm:ss")
    val timestamp: Date = Date()

    init {
        code = httpStatus.value()
        status = httpStatus.name
    }
}
